spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/stream/SkipperStreamDeployer.java

```diff
@@ -289,15 +292,23 @@ private void validateAllAppsRegistered(StreamDeploymentRequest streamDeploymentR
 			// nothing to validate.
 			return;
 		}
-
+		String streamName = streamDeploymentRequest.getStreamName();
 		// throw as at this point we should have definition
 		StreamDefinition streamDefinition = this.streamDefinitionRepository
-				.findById(streamDeploymentRequest.getStreamName())
+				.findById(streamName)
 				.orElseThrow(() -> new NoSuchStreamDefinitionException(streamDeploymentRequest.getStreamName()));
 
 		for (AppDeploymentRequest adr : streamDeploymentRequest.getAppDeploymentRequests()) {
+			String registeredAppName = getRegisteredName(streamDefinition, adr.getDefinition().getName());
+			String appName =  String.format("%s-%s-v", streamName, registeredAppName);
+			// Give 2 digits for version for now
+			if (appName.length()+2 > MAX_APPNAME_LENGTH) {
+				throw new InvalidStreamDefinitionException(
+						String.format("The runtime application name for the app %s in the stream %s "
+						+ "should not exceed %s in length. Currently it is: %s{version-2digits}", registeredAppName, streamName, MAX_APPNAME_LENGTH, appName));
+			}
 			String version = this.appRegistryService.getResourceVersion(adr.getResource());
-			validateAppVersionIsRegistered(getRegisteredName(streamDefinition, adr.getDefinition().getName()), adr, version);
+			validateAppVersionIsRegistered(registeredAppName, adr, version);
```

Comment:
`validateAppVersionIsRegistered` isn't a good method name any more, perhaps  `validateStreamDeploymentRequests` ?

Reply:
sorry, I meant the method `validateAllAppsRegistered` should be named `validateStreamDeploymentRequest`